account_sdk/src/session_token/account.rs

```diff
@@ -185,6 +164,27 @@ where
     }
 }
 
+pub fn parse_calls(execution: &RawExecution) -> Vec<Call> {
```

Comment:
Hm can we instead make these attributes public in starknet-rs? In the short term, we can use a fork

https://github.com/xJonathanLEI/starknet-rs/blob/c974e5cb42e8d8344cee910b76005ec46b4dd3ed/starknet-accounts/src/account/mod.rs#L126

Reply:
I've tried to use fork approach, there is a problem with recognizing `FieldElement` by the compiler.
`cainome` uses `starknet = "0.8.0"` as dependency.
I've tried fork with v0.8.0 tag and the commit you've provided.
The error message is such:
```
expected signature `fn(&SignatureProofs) -> Vec<starknet_ff::FieldElement>`
found signature `fn(&SignatureProofs) -> Vec<FieldElement>`
```

Reply:
I'm looking into this use case right now. Will revert shortly.

Btw regarding using a fork, you'll also need to patch the `starknet-ff` in a Cargo patch section for different `starknet-ff` dependencies to converge into a single crate. Simply deriving from the `v0.8.0` doesn't help, as Cargo treats crates from different sources as different.

Reply:
Just looked at the code in this PR. Please correct me if I'm wrong. @tarrencev @neotheprogramist

So far `starknet-rs` encapsulates everything in `RawExecution`, and only exposes a single `transaction_hash` function that allows computing the transaction hash. The expectation is simply that all sane account implementations would sign on transaction hashes instead of coming up with custom hashing schemes.

Looking at the code, it looks like the session account does _indeed_ work with transaction hash. However, on top of that, the account wraps the raw transaction hash and somehow compliments it with the list of raw calls to arrive at a final signature. So it needs access to the list of calls. And it would be nice if `starknet-rs` exposes it somehow.

Is this an accurate description of the problem? If so I can add access to the inner fields. I tend to make stuff hidden by default to preserve the ability to make changes without breaking downstream applications. But if there's a legitimate use case, I can definitely add it.

Reply:
Now done: https://github.com/xJonathanLEI/starknet-rs/pull/535.